src/instruction/instructions/logical/ands.ts

```diff
@@ -0,0 +1,66 @@
+import { IMemory } from 'board/memory/interfaces'
+import { Registers } from 'board/registers'
+import { ILabelOffsets } from 'instruction/interfaces'
+import {
+  checkOptionCount,
+  create,
+  createLowRegisterBits,
+  getBits,
+  isImmediate,
+  isOptionCountValid,
+  setBits
+} from 'instruction/opcode'
+import { Halfword, Word } from 'types/binary'
+import { evaluateZeroAndNegativeFlags } from '../../../board/alu'
+import { BaseInstruction } from '../base'
+
+/**
+ * Represents a 'Bitwise AND' instruction - ANDS
+ */
+export class AndsInstruction extends BaseInstruction {
+  public name: string = 'ANDS'
+  public pattern: string = '0100000000XXXXXX'
+  private rdnPattern: string = '0100000000000XXX'
+  private rmPattern: string = '0100000000XXX000'
+  private expectedOptionCount: number = 3
+
+  public canEncodeInstruction(name: string, options: string[]): boolean {
+    return (
+      super.canEncodeInstruction(name, options) &&
```

Comment:
Since there is only one implementation of AndsInstruction you cann ommit canEncodeInstruction here completly and let base class handle it by instruction name and do the additional checking as proposed by Leo in the encodeInstruction function

Reply:
Yes exactly this was my interpretation too. It is only really necessary if you have different classes for a instruction type.

Reply:
+1

I also vote for this 😄.

Reply:
Okay, I just fixed this.